Assignment2.yaml

```diff
@@ -0,0 +1,408 @@
+      summary: "Find Vehicle by ID"
+      description: "Returns a single vehicle"
+      produces:
+      - "application/json"
+      parameters:
+      - name: "VehicleId"
+        in: "path"
+        description: "ID of Vehicle to return"
+        required: true
+        type: "number"
+      - name: "q"
+        in: "query"
+        required: false
+        type: "string"
+        items:
+          type: "string"
+          enum:
+          - "name"
+          - "model"
+      responses:
+        200:
+          description: "successful operation"
+          schema:
+            $ref: "#/definitions/Vehicle"
+  /Vehicle/schema:
+    get:
+      tags:
+      - "Vehicle"
+      summary: "Find Vehicle Schema"
+      description: "Returns a json schema"
+      produces:
+      - "application/json"
+      responses:
+        200:
+          description: "successful operation"
+          schema:
+            $ref: "#/definitions/Vehicle"
+  /People:
+    get:
+      tags:
+      - "People"
+      summary: "Find list of People"
+      description: "get all the people resources"
+      produces:
+      - "application/json"
+      responses:
+        200:
+          description: "successful operation"
+          schema:
+            $ref: "#/definitions/people"
+  /People/{PeopleId}:
+    get:
+      tags:
+      - "People"
+      summary: "Find peoplee by ID"
+      description: "get a specific people resource"
+      produces:
+      - "application/json"
+      parameters:
+      - name: "PeopleId"
+        in: "path"
+        description: "ID of people to return"
+        required: true
+        type: "number"
+      - name: "q"
+        in: "query"
+        required: false
+        type: "string"
+        items:
+          type: "string"
+          enum:
+          - "name"
+      responses:
+        200:
+          description: "successful operation"
+          schema:
+            $ref: "#/definitions/people"
+  /People/schema:
+    get:
+      tags:
+      - "People"
+      summary: "Return Schema"
+      description: "Returns a json Schema"
+      produces:
+      - "application/json"
+      responses:
+        200:
+          description: "successful operation"
+          schema:
+            $ref: "#/definitions/people"
+  /starship:
+    get:
+      tags:
+      - "Starship"
+      summary: "Find list of starships"
+      description: "Returns a List of starship"
+      produces:
+      - "application/json"
+      responses:
+        200:
+          description: "successful operation"
+          schema:
+            $ref: "#/definitions/starship"
+  /starship/{starshipId}:
+    get:
+      tags:
+      - "Starship"
+      summary: "Find starship by ID"
+      description: "Returns a single starship"
+      produces:
+      - "application/json"
+      parameters:
+      - name: "starshipId"
+        in: "path"
+        description: "ID of starship to return"
+        required: true
+        type: "number"
+      - name: "q"
```

Comment:
I don't think `q` is a valid query param name. Even in the listing APIs the valid search param were 'name' or 'model'.